Automattic/jetpack
# Classic Theme Helper: Require Social Links from package in Jetpack and Jetpack Mu WPcom

Fixes https://github.com/Automattic/vulcan/issues/432
Fixes https://github.com/Automattic/vulcan/issues/433

## Proposed changes:

* This PR requires the Social Links package in the Jetpack plugin and the Jetpack Mu WPcom plugin, as well as adding in deprecations within the Jetpack module file.

### Other information:


## Jetpack product discussion

See linked issues.

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

 
Self-hosted and Atomic:
* On a test site, add a theme that includes support for Jetpack's social-links. The only theme I could find is 'Syntax'. 
* Otherwise, make sure you have a classic (not block) theme, such as Twenty Nineteen, and add the following snippet with a functionality plugin (like 'Code Snippets'):
```
add_action( 'after_setup_theme', function () {
	add_theme_support( 'social-links', array(
		'facebook', 'twitter', 'linkedin', 'tumblr',
	) );
} );
```
* Enable Social sharing - `/wp-admin/admin.php?page=jetpack#/sharing` and add a connection. 
* You can add a Facebook page by using the Gmail username and pw for `credsflow....` in the SS (internal only).
* Visit the Customizer, and you should see a 'Connect' menu item, that allows you to select your newly added connection.
* This should work both on trunk and with this PR applied.
* On Atomic make sure to test with the PR checked out for both Jetpack and WPcom Features, and each individually.
* You can also add an `error_log` in both the Jetpack module and Classic Theme Helper package files to confirm that it is the Jetpack file version that is loading (for Atomic I added to `jetpack-dev/jetpack-vendor/automattic/jetpack-classic-theme-helper...` as well as `jetpack-mu-wpcom-plugin/vendor/automattic/jetpack-classic-theme-helper...` to cover all bases.

Site Logo testing:
* I'd also recommend testing Site Logo as well, due to the seemingly unrelated issue flagged by phan and now with a phan suppression. Testing instructions are here: https://github.com/Automattic/jetpack/pull/37109
* I noticed no difference - checking the classes in the customizer for the uploader itself it's clear there is a subtle difference between custom-logo and site-icon (classname - `customize-control-site_logo` vs customize-control-custom_logo` - and it's location in the customizer).


Self-hosted only:
* You can check that there are no issues changing between the latest stable version of Jetpack (13.7), and this branch (no errors in logs).

Simple:
* Add a social connection at `https://wordpress.com/marketing/connections/yoursite.com`
* Add a theme which includes support for Jetpack's social-links - Syntax for example.
* Visit the Customizer and you should see the 'Connect' menu item allowing you to connect your new social connection.
* This should work both with and without the patches applied - see testing comment below (and also one at a time to replicate Jetpack only and Mu Wpcom only).
* As above, you can also add an `error_log` to confirm the location of the file being used - tailing errors with `tail -f /tmp/php-errors`.
